CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+cmake_minimum_required(VERSION 3.16)
+
+include(FetchContent)
+
+if (NOT CMAKE_TOOLCHAIN_FILE)
+    set(CMAKE_TOOLCHAIN_FILE
+        "${CMAKE_CURRENT_SOURCE_DIR}/build/emsdk/upstream/emscripten/cmake/Modules/Platform/\
+Emscripten.cmake"
+    )
+endif ()
+
+project(ClpIrStreamReader
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20
+# sources.
+if (NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR
+        "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif ()
+
+# Enable compile commands by default if the generator supports it.
+if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+        "Enable/Disable output of compile commands during generation." FORCE)
+endif ()
+
+# Set the default build type to Release if not specified
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    set(default_build_type "Release")
+    message(STATUS "No build type specified. Setting to '${default_build_type}'.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+endif ()
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0" CACHE STRING
+    "Version of Boost library to be used when compiling CLP sources.")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+set(BOOST_SRC_DIR "${CMAKE_CURRENT_BINARY_DIR}/boost" CACHE STRING
+    "Directory containing Boost sources, fetched from official release if non existent.")
+if (NOT EXISTS ${BOOST_SRC_DIR})
+    if (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.24)
+        # Sets DOWNLOAD_EXTRACT_TIMESTAMP=ON in FetchContent_Declare()
+        cmake_policy(SET CMP0135 NEW)
+    endif ()
+
+    FetchContent_Declare(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/\
+boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        SOURCE_DIR ${BOOST_SRC_DIR}
+    )
+    message(STATUS "Fetching Boost from official release.")
+    FetchContent_MakeAvailable(Boost)
+endif ()
+
+set(BIN_NAME "ClpIrStreamReader" CACHE STRING
```

Comment:
Is the plan to create different binaries for different sets of functionality or are we going to create a single binary for the entire library? If it's the latter, perhaps we should call this `ClpFfiJs`?

Reply:
I was thinking about releasing different binaries for different decoders (when IRv2 comes in) so that we can avoid loading unneeded binaries in the Log Viewer.
We can think more about the separation when IRv2 comes in. I agree `ClpFfijs` is more appropriate for now.